### ISSUE TITLE
Cannot use all filter options on certain fields because they are synced as fallback `type/*`

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#13141]
**Describe the bug**
Certain fields are synced as `type/*` instead of `type/Text` or `type/Number`, which means that only 4 filters are available `Is/Is not/Is empty/Not empty`, making it not possible to use regular filter options like "Contains".

**Workaround**
Use Custom Expression functions like `contains()`.

**To Reproduce**
I think it has something to do with TransactionTable, since I'm not seeing this behavior on GenericTable.

<details><summary>Sync log warnings</summary>

```
...
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'numeric(20,2)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'numeric(23,5)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(16383)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(256)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(16383)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'numeric(10,2)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'numeric(9,2)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(16383)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(2)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(24585)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'numeric(20,2)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(16383)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(13)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(16383)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(14)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(16383)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(2)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:41+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(3)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:42+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(16383)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:42+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(16383)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:42+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(14)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:42+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(16383)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:42+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(16386)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:42+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(2)' to a Field base_type, falling back to :type/*.
2020-08-19T16:27:42+08:00 WARN metabase.driver.sql-jdbc.sync Don't know how to map column type 'character varying(3)' to a Field base_type, falling back to :type/*.
...
```

</details>

<details><summary>Output of `/api/field/:id`</summary>

```
{
  "description": "My field",
  "database_type": "character varying(1000)",
  "table_id": 1300,
  "table": {
    "description": null,
    "entity_type": "entity/TransactionTable",
    "schema": "my_schema",
    "db": {
      ...
      "engine": "redshift",
      ...
    },
    "show_in_getting_started": false,
    "name": "my_table",
    "caveats": null,
    "updated_at": "2020-07-13T13:17:26.236261Z",
    "entity_name": null,
    "active": true,
    "id": 1300,
    "db_id": 2,
    "visibility_type": null,
    "field_order": "database",
    "display_name": "my_table",
    "created_at": "2020-04-16T06:30:48.957647Z",
    "points_of_interest": null
  },
  "special_type": "type/Category",
  "name": "my_field",
  "fingerprint_version": 4,
  "has_field_values": "search",
  "settings": null,
  "caveats": "None",
  "fk_target_field_id": null,
  "dimensions": [],
  "updated_at": "2020-08-18T13:05:19.724737Z",
  "custom_position": 0,
  "active": true,
  "parent_id": null,
  "id": 21587,
  "last_analyzed": "2020-04-16T06:47:45.326651Z",
  "position": 0,
  "visibility_type": "normal",
  "preview_display": true,
  "display_name": "my_field",
  "database_position": 12,
  "name_field": null,
  "fingerprint": {
    "global": {
      "distinct-count": 321,
      "nil%": 0
    }
  },
  "created_at": "2020-04-16T06:35:58.330572Z",
  "base_type": "type/*",
  "points_of_interest": null
}
```

</details>

**Information about your Metabase Installation:**
Metabase 1.35.4.1 and 1.36.3

**Comment by sbelak** [comment]
`entity_type` is only used in x-rays, so it should have no effect. It looks like we don't translate these parametrized field types to our vocabulary well.

**Comment by camsaul** [comment]
Fixed by #14496
